controllers/profiles/unfollowAnonUser.js

```diff
@@ -11,7 +11,13 @@ const UsersFunction = require('../../databases/functions/users');
 const {sendUnFollowMainFeedF2} = require('../../services/queue/mainFeedF2');
 
 module.exports = async (req, res) => {
-  const {user_id_followed, follow_source} = req.body;
+  let {user_id_followed, follow_source, post_id, comment_id} = req.body;
+
+  user_id_followed = await Getstream.feed.getUserIdFromSource(res, follow_source, {
+    post_id,
+    comment_id,
+    user_id: user_id_followed
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `Getstream.feed.getUserIdFromSource` function is being called without any error handling. If this function fails for any reason (e.g., the service is down, network issues), it could cause the application to crash or behave unexpectedly. It's a good practice to wrap such calls in a try-catch block to handle potential errors gracefully.

```diff
  let {user_id_followed, follow_source, post_id, comment_id} = req.body;

+ try {
    user_id_followed = await Getstream.feed.getUserIdFromSource(res, follow_source, {
      post_id,
      comment_id,
      user_id: user_id_followed
    });
+ } catch (error) {
+   return ErrorResponse.e500(res, 'Error while getting user ID from source');
+ }
```

This way, if there's an error, the server will respond with a 500 status code and a helpful error message, rather than crashing or hanging indefinitely.



<!-- This is an auto-generated comment by OSS CodeRabbit -->